server/app/views/questiontypes/ApplicantQuestionRendererImpl.java

```diff
@@ -78,23 +83,51 @@ public final DivTag render(ApplicantQuestionRendererParams params) {
         validationErrors.getOrDefault(question.getContextualizedPath(), ImmutableSet.of());
     if (!questionErrors.isEmpty()) {
       // Question error text
-      questionTextDiv.with(
+      questionSecondaryTextDiv.with(
           BaseHtmlView.fieldErrors(
               messages, questionErrors, ReferenceClasses.APPLICANT_QUESTION_ERRORS));
     }
 
     if (question.isRequiredButWasSkippedInCurrentProgram()) {
       String requiredQuestionMessage = messages.at(MessageKey.VALIDATION_REQUIRED.getKeyName());
-      questionTextDiv.with(
+      questionSecondaryTextDiv.with(
           div()
               .withClasses(Styles.P_1, Styles.TEXT_RED_600)
               .withText("*" + requiredQuestionMessage));
     }
 
+    // Composite fields should be rendered with fieldset and legend for screen reader users.
+    ContainerTag questionPrimaryTextTag;
+    ImmutableList<DomContent> questionTextDoms =
+        TextFormatter.createLinksAndEscapeText(
+            question.getQuestionText(), TextFormatter.UrlOpenAction.NewTab);
+    if (inputFieldType == InputFieldType.COMPOSITE) {
```

Comment:
nit: here and on L113 use `<KnownEnum>.equals(inputFieldType)`

Reply:
suggestion (non-blocking): here and L113 consider using a single `switch` statement instead of these two `if` statements

Reply:
Thanks! Updated to use a switch statement, and had to include a default case (otherwise I get a "variable questionPrimaryTextTag might not have been initialized" error)

I also changed the unit test to use a switch statement and exhaustively list out the enum types, since I think that makes the test clearer.

Reply:
great!